### DISCUSSION TITLE
how to set alarm on tracing for endpoint with err response code?

### DISCUSSION THREAD

**Discussion by houbinglei** [apache/skywalking#11358] [Q&A]
![image](https://github.com/apache/skywalking/assets/37292926/cb899bbc-94e6-478d-afda-ab6a303a4db2)

i have set endpoint oal metric:
endpoint_err_count = from(Endpoint.*).filter(httpResponseStatusCode in [400, 404, 500, 503]).count();

and i can get alert when /voiceBot/api... return 404; 
but how can i get alert when /rest/v2... return 404 and /voiceBot/api... return200?

**Comment by wu-sheng** [comment]
Alerting is only for metrics, not trace or log. If you want to set alerting, try to use OAL to extract metrics from traces.

**Reply by houbinglei** [reply]
Thanks very much for your detailed explanations!
i have tried this oal metric:
endpoint_err_count = from(Endpoint.*).filter(name in ["/rest/v2..."]).count();

but i  get the result with null; 

how could i  use OAL to extract metrics from traces?

**Reply by wu-sheng** [reply]
Check the endpoint page to get the endpoint name correctly. 
Don't guess.
